codice/alliance
# CAL-251 Adds support for the PIAPRD TRE

#### What does this PR do?
-Supports populating the metacard with PIAPRD TRE data when ingesting a NITF.
-It's possible for TRE fields to contain TRE groups (or indexed TRE attributes) within them - improves API to allow adding indexed TRE attributes to TRE fields so that data from the indexed TRE attributes can be populated in the metacard.
-Removes duplicate code in the NitfGmtiTransformer.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@harrison-tarr 
@dcruver
@bdeining 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@coyotesqrl
@jaymcnallie 
#### How should this be tested?
-Ingest a NITF containing the PIAPRD TRE and verify that the metacard created contains PIAPRD TRE data.
-Also ingest a NITF containing the MTIRPB TRE and verify that the metacard created contains MTIRPB TRE data. This TRE has a field that contains an indexed TRE attribute.
#### Any background context you want to provide?
#### What are the relevant tickets?
[CAL-251](https://codice.atlassian.net/browse/CAL-251)
#### Screenshots (if appropriate)
#### Checklist:
- [N/A] Update / Add Integration Tests
